app/models/solid_cache/entry.rb

```diff
@@ -33,7 +33,15 @@ def read(key)
 
       def read_multi(keys)
         without_query_cache do
-          find_by_sql([select_sql(keys), *key_hashes_for(keys)]).pluck(:key, :value).to_h
+          query = Arel.sql(select_sql(keys), *key_hashes_for(keys))
+
+          if SolidCache.configuration.encrypt?
+            # If encryption is enabled, we need to go through AR to decrypt the columns.
+            find_by_sql(query).pluck(:key, :value).to_h
+          else
+            # If encryption is disabled, we can go straight to the database.
+            connection.select_all(query, "SolidCache::Entry Load").cast_values.to_h
+          end
```

Comment:
```suggestion
          connection.select_all(query, "SolidCache::Entry Load").cast_values(SolidCache::Entry.attribute_types).to_h
```

Reply:
We can't rely on checking `SolidCache.configuration.encrypt?` because that just tells us if someone enabled encryption via Solid Cache. It's possible to enable it yourself via the run load hook.

But if we pass the attribute types in then it just works and we don't ever need to instantiate the AR objects 🎉